### PR TITLE
[DISCO-2295] Add Pull Request Template to Merino-py [do not deploy]

### PR DESCRIPTION
**Pull request by Trinaa**
## References

JIRA: [DISCO-2295](https://mozilla-hub.atlassian.net/browse/DISCO-2295)
GitHub: N/A

## Description

Proposal to add a PR template to this project. All ideas, modifications and suggestions welcome!

[Github - Creating a pull request template for your repository](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)

Examples\Inspo:

- [fxa template](https://github.com/mozilla/fxa/blob/main/.github/PULL_REQUEST_TEMPLATE.md)
- firefox-ios
  - [template](https://github.com/mozilla-mobile/firefox-ios/blob/main/PULL_REQUEST_TEMPLATE)
  - [Pull Request Naming Guide](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [firefox-android template](https://github.com/mozilla-mobile/firefox-android/blob/main/.github/pull_request_template.md)
- [experimenter template](https://github.com/mozilla/experimenter/blob/main/PULL_REQUEST_TEMPLATE.md)
- [blurts-server template](https://github.com/mozilla/blurts-server/blob/main/.github/pull_request_template.md)

## DoD Checklist

_Put an `x` in the boxes that apply_

- [ ] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
- [x] `[do not deploy]` and `[load test: (abort|warn)]` keywords are applied (if applicable)
- [x] Documentation has been updated (if applicable)
- [x] [Functional and performance test](https://github.com/mozilla-services/merino-py/blob/main/docs/dev/testing.md) coverage has been expanded and maintained (if applicable)

[DISCO-2295]: https://mozilla-hub.atlassian.net/browse/DISCO-2295?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## References
+
+JIRA: [DISCO-TODO](https://mozilla-hub.atlassian.net/browse/DISCO-TODO)
+Github: [#TODO](https://github.com/mozilla-services/merino-py/issues/TODO)
```
Comment (linabutler) [inline PULL_REQUEST_TEMPLATE.md:4]:
```suggestion
GitHub: [#TODO](https://github.com/mozilla-services/merino-py/issues/TODO)
```

(Sorry! 😅)

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+<!-- Detail the reason and impact of this PR, along with any other relevant information including: change highlights, screenshots, test instructions, etc .... -->
+
+
+
+## DoD Checklist
+
+_Put an `x` in the boxes that apply_
+
+- [ ] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
+- [ ] \[do not deploy\] and \[load test: (abort|warn)\] labels are applied (if applicable)
```
Comment (linabutler) [inline PULL_REQUEST_TEMPLATE.md:16]:
```suggestion
- [ ] `[do not deploy]` and `[load test: (abort|warn)]` labels are applied (if applicable)
```

**Review by linabutler — APPROVED** [review]
This looks lovely, thank you for doing this!

**Review by ncloudioj — APPROVED** [review]
LGTM, thanks! `[deploy granted]`

**Review by taddes — APPROVED** [review]
Great stuff @Trinaa , I added a couple comments for your consideration, thanks for doing this.

**Review by tiftran — APPROVED** [review]
Thank you for doing this!

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+## References
+
+JIRA: [DISCO-TODO](https://mozilla-hub.atlassian.net/browse/DISCO-TODO)
+GitHub: [#TODO](https://github.com/mozilla-services/merino-py/issues/TODO)
+
+## Description
+<!-- Detail the reason and impact of this PR, along with any other relevant information including: change highlights, screenshots, test instructions, etc .... -->
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply PULL_REQUEST_TEMPLATE.md]: - I can shift the order, I can think of a few examples though where a change list is perhaps not necessary, so perhaps leave that to the discretion of the PR author.
- I will add an item for PR title format to the checklist

File: PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+## References
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply PULL_REQUEST_TEMPLATE.md:1]: - I think the PR title, as you mention bellow, should ideally be exact or synonymous with the JIRA ticket. I will add a checklist item.

**Pull request merged**